src/java.base/share/native/libjli/java.c

```diff
@@ -328,29 +366,17 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argv */
         }
     }
 
+    /* Process -Xmso in the options.default file, the JAVA_TOOL_OPTIONS, OPENJ9_JAVA_OPTIONS
+     * and IBM_JAVA_OPTIONS environment variables, to set the main thread stack size. May be
+     * overridden by a later command line option.
+     */
     {
-        /* Process -Xmso in the OPENJ9_JAVA_OPTIONS environment variable to
-         * set the main thread stack size. May be overridden by a later command
-         * line option.
-         */
-        JLI_List openj9Args = JLI_List_new(8); /* 8 is arbitrary */
-        if (JLI_ParseOpenJ9ArgsFromEnvVar(openj9Args, "OPENJ9_JAVA_OPTIONS")) {
-            size_t i = openj9Args->size;
-            while (i > 0) {
-                i -= 1;
-                if (JLI_StrCCmp(openj9Args->elements[i], "-Xmso") == 0) {
-                    jlong tmp = 0;
-                    if (parse_size(openj9Args->elements[i] + 5, &tmp)) {
-                        threadStackSize = tmp;
-                        if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
-                            threadStackSize = STACK_SIZE_MINIMUM;
-                        }
-                    }
-                    break;
-                }
-            }
-            JLI_List_free(openj9Args);
-        }
+        char optionsfile[MAXPATHLEN*2]; /* The size needs to be bigger than jrepath to avoid a warning. */
```

Comment:
We shouldn't assume a relationship between `MAXPATHLEN` and the length of the suffix we are adding:
```c
        char optionsfile[sizeof(jrepath) + sizeof("/lib/options.default") - 1];
```